components/cluster/command/display.go

```diff
@@ -54,20 +57,24 @@ func newDisplayCmd() *cobra.Command {
 				return perrs.Errorf("Cluster %s not found", clusterName)
 			}
 
+			metadata, err := spec.ClusterMetadata(clusterName)
+			if err != nil && !errors.Is(perrs.Cause(err), meta.ErrValidate) &&
+				!errors.Is(perrs.Cause(err), spec.ErrNoTiSparkMaster) {
+				return perrs.AddStack(err)
+			}
 			if showDashboardOnly {
-				return displayDashboardInfo(clusterName)
+				tlsCfg, err := metadata.Topology.TLSConfig(tidbSpec.Path(clusterName, spec.TLSCertKeyDir))
+				if err != nil {
+					return perrs.AddStack(err)
```

Comment:
It's better to add a stack in the function `metadata.Topology.TLSConfig` to avoid `perrs.AddStack` everywhere.

Reply:
It's a wrapped direct returning of etcd client's function, no named error variable in the call stack.